scripts/host_explore.py

```diff
@@ -44,7 +44,7 @@ def check_output(command, dir=None):
         output = subprocess.check_output(command, cwd=dir).strip()
         output = output.decode(sys.getdefaultencoding())
     except OSError as e:
-        logger.error(str(e))
+        logger.error("%s executing '%s'" % (str(e), command))
```

Comment:
`command` is an array usually - so this won't be very pretty, although it should still work.

You can do something like `" ".join([i for i in pipes.quote(i)])`? Although it is possible overkill :)

Reply:
Well if it's always a list then:
```suggestion
        logger.error("%s executing '%s'" % (str(e), ' '.join(command)))
```
if it can be string also, then it require one more check:
```suggestion
        logger.error("%s executing '%s'" % (str(e),'{}'.format(' ' if isinstance(x
command, list) else '').join(command)))
```

Reply:
command is always a list, as this is what subprocess.check_output expects

Rather than import a separate module, I'm just printing command[0]. This should be enough to identify why we're getting an OSError. Problems encountered by the command should result in the CalledProcessError instead, and that should appropriately escape the command.

Reply:
Sure, that works